numpy/core/memmap.py

```diff
@@ -269,7 +269,7 @@ def __new__(subtype, filename, dtype=uint8, mode='r+', offset=0,
             self.offset = offset
             self.mode = mode
 
-            if is_pathlib_path(filename):
```

Comment:
This change isn't safe. because it uses `resolve` below instead of `os_fspath` - pathlike objects aren't required to implement `resolve`

Reply:
So os_PatgLike is a subset of pathlike?

Reply:
`os.PathLike` is a subset of `pathlib.Path`, if that was what you were asking

Reply:
ah i see

Reply:
```
In [1]: from pathlib import Path                                                

In [2]: import os.path                                                          

In [3]: os.path                                                                 
Out[3]: <module 'posixpath' from '/home/mark/miniconda3/envs/mcam_dev/lib/python3.7/posixpath.py'>

In [4]: os.path.abspath(Path())                                                 
Out[4]: '/home/mark/git/numpy'
``` 

should be ok I think

Reply:
Apparently only in python 3.8 +  https://bugs.python.org/issue9949

Reply:
You're looking for `filename = os_fspath(filename)` in this branch - but that becomes a feature, so probably belongs in its own PR